### PR TITLE
DOC: split quantization.rst into smaller pieces

### PR DESCRIPTION
**Pull request by mattip**
xref gh-38010 and gh-38011.

After this PR, there should be only two warnings:
```
pytorch/docs/source/index.rst:65: WARNING: toctree contains reference to nonexisting \
      document 'torchvision/index'
WARNING: autodoc: failed to import class 'tensorboard.writer.SummaryWriter' from module \
     'torch.utils'; the following exception was raised:
No module named 'tensorboard'
```

If tensorboard and torchvision are prerequisites to building docs, they should be added to the `requirements.txt`.

As for breaking up quantization into smaller pieces: I split out the list of supported operations and the list of modules to separate documents. I think this makes the page flow better, makes it much "lighter" in terms of page cost, and also removes some warnings since the same class names appear in multiple sub-modules.

### CODE REVIEW

File: docs/source/jit.rst
Comment (mattip) [inline docs/source/jit.rst:292]: these two labels are needed for references in jit._script.py

File: docs/source/type_info.rst
Comment (mattip) [inline docs/source/type_info.rst:31]: The addition of the "resolution" row broke formatting. This happened sometime in the past week, making the case stronger for not allowing warnings to pass, gh-38011

File: torch/csrc/api/include/torch/nn/modules/loss.h
Comment (mattip) [inline torch/csrc/api/include/torch/nn/modules/loss.h:56]: The tag line of the C++ function ends up [in the documentation](https://pytorch.org/docs/master/nn.html#loss-functions) (in the table under that link). No need to use reference formatting, since the link is in the code and [the full docstring](https://pytorch.org/docs/master/generated/torch.nn.KLDivLoss.html#torch.nn.KLDivLoss) properly formats the link.

File: torch/jit/_script.py
Comment (rgommers) [inline torch/jit/_script.py:421]: This looks incorrect, and results in broken links (see https://6214508-65600975-gh.circle-artifacts.com/0/docs/generated/torch.jit.ScriptModule.html):

image

you may want `:ref:`?

Comment (rgommers) [reply torch/jit/_script.py:421]: and same for the reference just below

**Review by rgommers — APPROVED** [review]
LGTM modulo the one broken link issue. The main `quantization` page looks a lot better.

**Comment by rgommers** [comment]
> If tensorboard and torchvision are prerequisites to building docs, they should be added to the `requirements.txt`.

I would move this conversation to gh-38011.

File: torch/jit/_script.py
Comment (mattip) [reply torch/jit/_script.py]: fixed

**Comment by gottbrath** [comment]
Folks. I added Raghu and  Vasiliy as reviewers for this PR. They are the content matter experts on quantization and we should get their input on the impact of this before landing this change.

**Comment by rgommers** [comment]
> They are the content matter experts on quantization and we should get their input on the impact of this before landing this change.

There are no significant changes in content, so I'd recommend landing this quickly to finally get CI to ensure we stay at zero doc build warnings. @raghuramank100 or @vkuzo, would you be able to look at this soon-ish?

**Review by vkuzo — APPROVED** [review]
the quantization part LGTM.  Thanks for doing this, it should read much better after this change.

**Review by mruberry — APPROVED** [review]
Cool!

**Pull request closed without merging**
